packages/components/src/list/experimental-list-item.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { CSSTransition } from 'react-transition-group';
+
+/**
+ * Internal dependencies
+ */
+import { handleKeyDown } from './list-item';
+
+type CSSTransitionProps = {
+	in: boolean;
+	exit: boolean;
+	enter: boolean;
+	onExited: () => void;
+};
+
+type ListItemProps = {
+	onClick?: () => void;
+	// control whether to display padding on list item or not.
+	disableGutters: boolean;
+	className?: string;
+	// By default the List is a `nav`, which treats the list item as role `link` if `onClick` is passed. To override this pass a different role.
+	role?: 'menuitem' | 'link' | 'button';
+	// By default a div is rendered, but if you want the list item to behave as a different tag you can override it here.
+	component?: React.ElementType;
+} & CSSTransitionProps &
+	React.HTMLAttributes< HTMLElement >;
+
+export const ExperimentalListItem: React.FC< ListItemProps > = ( {
+	children,
+	disableGutters = false,
+	className = '',
+	role,
+	component,
+	tabIndex,
+	// extract out the props that must be passed down from TransitionGroup
+	exit,
+	enter,
+	onExited,
+	// in is a TS reserved keyword so can't be a variable name
+	in: transitionIn,
+
+	// Everything else you might pass into an HTML element
+	...otherProps
+} ): JSX.Element => {
+	const hasAction = !! otherProps.onClick;
+	const tagRole = role || ( component !== 'a' && hasAction && 'link' ) || '';
+	const TagName = component || 'div';
+	const tagTabIndex = tabIndex || ( hasAction ? '0' : null );
+
```

Comment:
Something seems a bit off in this logic. Here is my opinion:

* The `tagName` should always be `a` if an `href` is supplied, no need for `component === 'a'`
* The `onClick` should always render a `button` tag, unless there is an `href`. Creating a `div` with role `a` is a roundabout method that prevents the browser from displaying a preview of the link. It also assumes clicking the item results in a navigation. I don't think we should be guiding developers toward navigating in an `onClick`

Reply:
@psealock I agree with all of this. I went back and forward on these so its great to have some feedback. I'll implement this.

Reply:
I think this is resolved in https://github.com/woocommerce/woocommerce-admin/pull/6787/files#r614489777 keen to hear thoughts.